pkg/cache/v3/cache.go

```diff
@@ -83,299 +116,44 @@ type Cache interface {
 
 // Response is a wrapper around Envoy's DiscoveryResponse.
 type Response interface {
-	// Get the Constructed DiscoveryResponse
-	GetDiscoveryResponse() (*discovery.DiscoveryResponse, error)
-
-	// Get the original Request for the Response.
-	GetRequest() *discovery.DiscoveryRequest
+	GetRequest() Request
 
 	// Get the version in the Response.
 	GetVersion() (string, error)
 
+	// Get the list of resources included in the response.
+	GetReturnedResources() []string
+
 	// Get the context provided during response creation.
 	GetContext() context.Context
+
+	// Get the Constructed DiscoveryResponse
```

Comment:
Imo those should not be in Cache, as those depend on the server and the cache does not really need to know what requests/responses are passed
For now I kept them there to limit the impact